url64d634b8f0_agdturner_generic_tgz-pJ8-uk_ac_leeds_ccg_andyt_generic_visualisation_charts_Generic_BarChartJ8/src/uk/ac/leeds/ccg/andyt/generic/visualisation/charts/Generic_Plot.java

```diff
@@ -0,0 +1,248 @@
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package uk.ac.leeds.ccg.andyt.generic.visualisation.charts;
+
+import java.awt.Rectangle;
+import java.awt.print.PrinterJob;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.util.concurrent.Future;
+import javax.print.StreamPrintService;
+import javax.print.StreamPrintServiceFactory;
+import javax.print.attribute.HashPrintRequestAttributeSet;
+import javax.print.attribute.PrintRequestAttributeSet;
+import javax.print.attribute.standard.Copies;
+import uk.ac.leeds.ccg.andyt.generic.utilities.Generic_Execution;
+import uk.ac.leeds.ccg.andyt.generic.visualisation.Generic_Canvas;
+import uk.ac.leeds.ccg.andyt.generic.visualisation.Generic_Printable;
+import uk.ac.leeds.ccg.andyt.generic.visualisation.Generic_Visualisation;
+
+/**
+ * This class acts holds a reference to a lightweight component for rendering.
+ * In part it acts as an interface and is extended by other classes for
+ * generating specific data plots.
+ */
+public class Generic_Plot extends Abstract_Generic_Plot {
+
+    public Generic_Canvas _Generic_Canvas;
+    public Future future;
+
+    public Generic_Plot() {
+    }
+
+    /**
+     *
+     */
+    @Override
+    public void run() {
+        try {
+            File theFile = getFile();
+            String theFormat = getFormat();
+            _Generic_Canvas = new Generic_Canvas();
+            _Generic_Canvas._Generic_Plot = this;
+            _Generic_Canvas._Rectangle = new Rectangle(0, 0, getDataWidth(), getDataHeight());
+
+//        Graphics2D g2 = (Graphics2D) _Generic_Canvas.getGraphics();
+//            setG2(g2);
+//            //_Generic_Canvas.paint(g2);
+//            Dimension dim = draw();
+//            _Generic_Canvas.setDimension(dim);
+            PrinterJob printerJob = PrinterJob.getPrinterJob();
+            Generic_Printable theGeneric_Printable = new Generic_Printable(_Generic_Canvas);
+            printerJob.setPrintable(theGeneric_Printable);
+
+            String psMimeType = "application/postscript";
+
+            FileOutputStream fileOutputStream = null;
+//        PrintService[] printServices = PrinterJob.lookupPrintServices();
+            StreamPrintService streamPrintService = null;
+            StreamPrintServiceFactory[] streamPrintServiceFactories
+                    = PrinterJob.lookupStreamPrintServices(psMimeType);
+            File psFile = new File(theFile.getParentFile(), theFile.getName() + ".ps");
+            System.out.println("psFile " + psFile.toString());
+            if (streamPrintServiceFactories.length > 0) {
+                try {
+                    psFile.createNewFile();
+                    fileOutputStream = new FileOutputStream(psFile);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

On this line, psFile is used as input of method call FileOutputStream() to compute fileOutputStream. If this method call throws an exception, the value of fileOutputStream is not updated, thus, the states of psFile and fileOutputStream might be inconsistent when used later. Consider verifying the value of fileOutputStream after calling FileOutputStream() to ensure the value is what you expect it to be.